### ISSUE TITLE
[KED-2897] `kedro pipeline pull` doesn't work if `requirements.txt` contains `-r` or `-c` entries

### ISSUE THREAD

**Issue by deepyaman** [quantumblacklabs/kedro#913]
## Description
`kedro pipeline pull` doesn't work if `requirements.txt` contains `-r` or `-c` entries. `requirements.txt` gets copied without modification to `requirements.in` if it doesn't already exist, including the `-r`/`-c` flags.

## Context
I have a constraints file, and I have to work around it if I want to be able to pull in packages.

## Steps to Reproduce
1. Create a Kedro project
2. Add a line like `-c constraints.txt` or `-r more-requirements.txt` to `src/requirements.txt`
3. `kedro pipeline pull` something

## Expected Result
It should process the linked constraints/requirements.

## Actual Result
https://github.com/quantumblacklabs/kedro/blob/0.17.5/kedro/framework/cli/pipeline.py#L1107 results in an error:

```python
Traceback (most recent call last):
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/bin/kedro", line 8, in <module>
    sys.exit(main())
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/kedro/framework/cli/cli.py", line 266, in main
    cli_collection()
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/kedro/framework/cli/cli.py", line 216, in main
    **extra,
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/kedro/framework/cli/pipeline.py", line 311, in pull_package
    _pull_package(package_path, metadata, env=env, alias=alias, fs_args=fs_args)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/kedro/framework/cli/pipeline.py", line 349, in _pull_package
    _append_package_reqs(requirements_in, package_reqs, package_name)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/kedro/framework/cli/pipeline.py", line 1107, in _append_package_reqs
    reqs_to_add = set(new_reqs) - set(existing_reqs)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3084, in parse_requirements
    yield Requirement(line)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3094, in __init__
    super(Requirement, self).__init__(requirement_string)
  File "/Users/deepyaman_datta/miniconda3/envs/customerone-retail/lib/python3.7/site-packages/pkg_resources/_vendor/packaging/requirements.py", line 102, in __init__
    requirement_string[e.loc : e.loc + 8], e.msg
pkg_resources.extern.packaging.requirements.InvalidRequirement: Parse error at "'-c const'": Expected W:(abcd...)
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): kedro, version 0.17.5
* Python version used (`python -V`): Python 3.7.11
* Operating system and version:
  ```
  ProductName:	macOS
  ProductVersion:	11.5.1
  BuildVersion:	20G80
  ```

**Comment by deepyaman** [comment]
It also doesn't clean up (i.e. you're left with the exploded pipeline as a side effect), even though the command failed.

**Comment by AntonyMilneQB** [comment]
This will be fixed in 0.17.6. Thank you @deepyaman!
